File: tests/unit/java/com/trulioo/sdk/model/BusinessRegistrationNumberTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;

public class BusinessRegistrationNumberTest {
    private BusinessRegistrationNumber businessRegistrationNumber;

    @BeforeEach
    public void beforeEach() {
        businessRegistrationNumber = new BusinessRegistrationNumber();
    }

    @Test
    public void testName() {
        String name = "test";
        businessRegistrationNumber.setName(name);

        assertEquals(name, businessRegistrationNumber.getName());
        assertEquals(name, new BusinessRegistrationNumber().name(name).getName());
    }

    @Test
    public void testDescription() {
        String description = "desc";
        businessRegistrationNumber.setDescription(description);

        assertEquals(description, businessRegistrationNumber.getDescription());
        assertEquals(description, new BusinessRegistrationNumber().description(description).getDescription());
    }

    @Test
    public void testCountry() {
        String country = "Canada";
        businessRegistrationNumber.setCountry(country);

        assertEquals(country, businessRegistrationNumber.getCountry());
        assertEquals(country, new BusinessRegistrationNumber().country(country).getCountry());
    }

    @Test
    public void testJurisdiction() {
        String jurisdiction = "Alberta";
        businessRegistrationNumber.setJurisdiction(jurisdiction);

        assertEquals(jurisdiction, businessRegistrationNumber.getJurisdiction());
        assertEquals(jurisdiction, new BusinessRegistrationNumber().jurisdiction(jurisdiction).getJurisdiction());
    }

    @Test
    public void testSupported() {
        Boolean supported = true;
        businessRegistrationNumber.setSupported(supported);

        assertEquals(supported, businessRegistrationNumber.getSupported());
        assertEquals(supported, new BusinessRegistrationNumber().supported(supported).getSupported());
    }

    @Test
    public void testType() {
        String type = "type";
        businessRegistrationNumber.setType(type);

        assertEquals(type, businessRegistrationNumber.getType());
        assertEquals(type, new BusinessRegistrationNumber().type(type).getType());
    }

    @Test
    public void testMasks() {
        businessRegistrationNumber.addMasksItem(new BusinessRegistrationNumberMask());
        businessRegistrationNumber.addMasksItem(new BusinessRegistrationNumberMask());
        List<BusinessRegistrationNumberMask> masks = Arrays.asList(new BusinessRegistrationNumberMask().mask("AAA"));
        businessRegistrationNumber.setMasks(masks);

        assertEquals(masks, businessRegistrationNumber.getMasks());
        assertEquals(masks, new BusinessRegistrationNumber().masks(masks).getMasks());
    }

    @Test
    public void testEquals() {
        String name = "test";
        BusinessRegistrationNumber businessRegistrationNumber1 = new BusinessRegistrationNumber().name(name);

        assertEquals(businessRegistrationNumber1, businessRegistrationNumber1);
        assertEquals(businessRegistrationNumber1, new BusinessRegistrationNumber().name(name));
        assertNotEquals(businessRegistrationNumber1, new BusinessRegistrationNumber().name(name + "1"));
        assertNotEquals(businessRegistrationNumber1, null);
    }

    @Test
    public void testHashcode() {
        String name = "test";

        assertEquals(new BusinessRegistrationNumber().name(name).hashCode(),
                new BusinessRegistrationNumber().name(name).hashCode());
    }

    @Test
    public void testToString() {
        String name = "test";

        assertEquals(new BusinessRegistrationNumber().name(name).toString(),
                new BusinessRegistrationNumber().name(name).toString());
    }
}

File: tests/unit/php/Model/BusinessRegistrationNumberTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use InvalidArgumentException;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\BusinessRegistrationNumber;
use Trulioo\SDK\Model\BusinessRegistrationNumberMask;

class BusinessRegistrationNumberTest extends TestCase
{
    private BusinessRegistrationNumber $businessRegistrationNumber;

    public function setUp(): void
    {
        $this->businessRegistrationNumber = new BusinessRegistrationNumber([]);
    }

    public function testPropertyName()
    {
        $this->businessRegistrationNumber->setName('test');
        $this->assertEquals('test', $this->businessRegistrationNumber->getName());

        try {
            $this->businessRegistrationNumber->setName(str_repeat('a', 513));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyDescription()
    {
        $this->businessRegistrationNumber->setDescription('desc');
        $this->assertEquals('desc', $this->businessRegistrationNumber->getDescription());

        try {
            $this->businessRegistrationNumber->setDescription(str_repeat('a', 513));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyCountry()
    {
        $this->businessRegistrationNumber->setCountry('Canada');
        $this->assertEquals('Canada', $this->businessRegistrationNumber->getCountry());

        try {
            $this->businessRegistrationNumber->setCountry(str_repeat('a', 513));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyJurisdiction()
    {
        $this->businessRegistrationNumber->setJurisdiction('Alberta');
        $this->assertEquals('Alberta', $this->businessRegistrationNumber->getJurisdiction());

        try {
            $this->businessRegistrationNumber->setJurisdiction(str_repeat('a', 513));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertySupported()
    {
        $this->businessRegistrationNumber->setSupported(true);
        $this->assertEquals(true, $this->businessRegistrationNumber->getSupported());
    }

    public function testPropertyType()
    {
        $this->businessRegistrationNumber->setType('type');
        $this->assertEquals('type', $this->businessRegistrationNumber->getType());

        try {
            $this->businessRegistrationNumber->setType(str_repeat('a', 513));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyMasks()
    {
        $this->businessRegistrationNumber->setMasks([
            new BusinessRegistrationNumberMask([
                'mask' => 'aaa'
            ])
        ]);
        $this->assertEquals(
            [
                new BusinessRegistrationNumberMask([
                    'mask' => 'aaa'
                ])
            ],
            $this->businessRegistrationNumber->getMasks()
        );
    }
}
